apps/ubuntu_bootstrap/lib/pages/storage/storage_guided_capabilities_page.dart

```diff
@@ -0,0 +1,196 @@
+                      OptionButton(
+                        title: Text(lang.installationTypeLVM),
+                        isThreeLines: false,
+                        value: GuidedCapability.LVM,
+                        groupValue: model.guidedCapability,
+                        onChanged: (v) => model.guidedCapability = v!.clean(),
+                      ),
+                    if (model.hasZfs) ...[
+                      OptionButton(
+                        title: Wrap(
+                          children: [
+                            Text(lang.installationTypeZFS),
+                            InfoBadge(title: experimentalBadgeText),
+                          ],
+                        ),
+                        isThreeLines: false,
+                        value: GuidedCapability.ZFS,
+                        groupValue: model.guidedCapability,
+                        onChanged: (v) => model.guidedCapability = v!.clean(),
+                      ),
+                      OptionButton(
+                        title: Wrap(
+                          children: [
+                            Text(lang.installationTypeZFSEncryption),
+                            InfoBadge(title: experimentalBadgeText),
+                          ],
+                        ),
+                        isThreeLines: false,
+                        value: GuidedCapability.ZFS_LUKS_KEYSTORE,
+                        groupValue: model.guidedCapability,
+                        onChanged: (v) => model.guidedCapability = v!.clean(),
+                      ),
+                    ],
+                    TpmOption(
+                      model: model,
+                    ),
+                  ].withSpacing(kWizardSpacing / 2),
+                ),
+              ),
+              Visibility(
+                visible: !model.showAdvanced,
+                child: Column(
+                  children: [
+                    const SizedBox(height: kWizardSpacing),
+                    OutlinedButton(
+                      onPressed: model.toggleShowAdvanced,
+                      child: Text(
+                        lang.installationTypeAdvancedLabel,
+                      ),
+                    ),
+                  ],
+                ),
+              ),
+            ],
+          ),
+        ),
+      ],
+    );
+  }
+}
+
+class TpmOption extends ConsumerWidget {
+  const TpmOption({
+    required this.model,
+    super.key,
+  });
+  final StorageModel model;
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final lang = UbuntuBootstrapLocalizations.of(context);
+    final flavor = ref.watch(flavorProvider);
+    final experimentalBadgeText =
+        UbuntuBootstrapLocalizations.of(context).installationTypeExperimental;
+
+    final target = model
+        .getAllTargets()
+        .firstWhereOrNull((t) => t is GuidedStorageTargetReformat);
+
+    if (target == null) return const SizedBox();
```

Comment:
We will need to generalize this logic here, since we can now see this option while a different guided target (one that is not `GuidedStorageTargetReformat`) is selected.

I've tested this with the `win10-along-ubuntu` machine config, where we get a `Reformat` and two `Resize` targets. The latter don't include the `CORE_BOOT_PREFER_ENCRYPTED` capability in neither `allowed` nor `disallowed`, so I think we should hide the option entirely in this case (currently it's shown and enabled, so continuing with the installation will lead to a crash, since we're sending a guided target response that includes an unsupported capability). At the same time  we should make sure that the option is always present if it's included in one of those, but keep it disabled if the capability is disallowed.